spi/data-plane/data-plane-spi/src/main/java/org/eclipse/edc/connector/dataplane/spi/pipeline/StreamFailure.java

```diff
@@ -29,6 +30,16 @@ public StreamFailure(List<String> messages, Reason reason) {
         this.reason = reason;
     }
 
+    @Override
+    public String getFailureDetail() {
+        var str = super.getFailureDetail();
+        if (str != null && !str.isEmpty()) {
+            str = reason + ": " + str;
+            return str;
+        }
+        return String.valueOf(reason);
+    }
+
```

Comment:
This method can be simplified using the ternary operator as part of a return statement and eliminating the `String.valueOf()` call:

`return str != null && !str.isEmpty() ? ....`